lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -1878,15 +1875,134 @@ class SqlParser(private val ion: IonSystem) : Parser {
         }
     }
 
+    private fun List<Token>.parseFromSource(precedence: Int = -1, parseRemaining: Boolean = true): ParseNode {
+        var rem = this
+        var child = when (rem.head?.keywordText) {
+            "unpivot" -> {
+                val actualChild = rem.tail.parseExpression(precedence)
+                ParseNode(
+                        UNPIVOT,
+                        rem.head,
+                        listOf(actualChild),
+                        actualChild.remaining
+                )
+            }
+            else -> {
+                val isSubqueryOrLiteral = rem.tail.head?.type == LITERAL || rem.tail.head?.keywordText == "select"
+                if (rem.head?.type == LEFT_PAREN && !isSubqueryOrLiteral) {
+                    // Starts with a left paren and is not a subquery or literal, so parse as a from source
+                    rem = rem.tail
+                    rem.parseFromSource(precedence).deriveExpected(RIGHT_PAREN)
+                }
+                else {
+                    rem.parseExpression(precedence)
+                }
+            }
+        }
+        rem = child.remaining
+
+        child = rem.parseOptionalAsAlias(child).also {
+            rem = it.remaining
+        }
+
+        child = rem.parseOptionalAtAlias(child).also {
+            rem = it.remaining
+        }
+
+        child = rem.parseOptionalByAlias(child).also {
+            rem = it.remaining
+        }
+
+        var left = child
+
+        var delim = rem.parseJoinDelim()
+        if (parseRemaining) {
+            while (delim?.type?.isJoin == true) {
+                val isCrossJoin = delim.token?.keywordText?.contains("cross") ?: false
+                val hasOnClause = delim.token?.type == KEYWORD && !isCrossJoin
+                var children : List<ParseNode>
+                var joinToken : Token? = delim.token
+
+                rem = rem.tail
+
+                if (hasOnClause) {
+                    // Explicit join
+                    if (rem.head?.type == LEFT_PAREN) {
+                        // Starts with a left paren. Could indicate subquery/literal or indicate higher precedence
+                        val isSubqueryOrLiteral = rem.tail.head?.type == LITERAL || rem.tail.head?.keywordText == "select"
+                        val parenClause = rem.parseFromSource(precedence, true)
```

Comment:
use named arguments for when passing boolean literals.

Reply:
Changed to use named argument for the boolean literal.